src/database/DataTypes/StocktakeBatch.js

```diff
@@ -115,11 +115,45 @@ export class StocktakeBatch extends Realm.Object {
   }
 
   /**
-   * Returns if this batch has a reason applied. Simple getter
-   * matching StocktakeItem interface for ease of use.
+   * Indicator that this batch has a positive difference in snapshot and
+   * counted quantity.
    */
-  get hasReason() {
-    return !!this.option;
+  get hasPositiveAdjustment() {
+    return this.difference > 0;
+  }
+
+  /**
+   * Indicator that this batch has a negative difference in snapshot and
+   * counted quantity.
+   */
+  get hasNegativeAdjustment() {
+    return this.difference < 0;
+  }
+
+  /**
+   * Returns an indicator that this batches reason/option state is valid.
+   * Valid being: negative differences require a negativeInventoryAdjustment
+   * option. Positive differences require a positiveInventoryAdjustment option
+   * while no difference requires there to be no option applied.
+   */
+  get validateReason() {
+    const { difference, option, hasPositiveAdjustment, hasNegativeAdjustment } = this;
+
+    // Short circuits for simple cases
+    if (!difference && !option) return true;
+    if (difference && !option) return false;
+
+    // Determine the validity of the reason state where there is a difference
+    // and the batch already has a option.
+    const { type } = option;
+
```

Comment:
I want to make a database constants file. I will make an issue for it, I think it would be really helpful. I hate TODO's a lot, it's equivalent to just doing nothing, which I think is what i'm gonna do here (I reallly wanna do the DB constants file though)

Reply:
I disagree on that one, `TODO`s are only equivalent to nothing if you use them as an excuse to not do anything. I've seen then used in a few projects for marking areas of code to other programmers + warning them of code that needs improvement (issues can do this, but often TODOs are combined with issues). They also provide a metric for technical debt; when something is "hacked", I think it's a good idea to leave a `TODO`. Then, if your TODOs are piling up, you know you need to start fixing them.

Since its a high priority, fine to leave it here, though.